fe/fe-core/src/test/java/com/starrocks/sql/plan/StructTypeTest.java

```diff
@@ -41,19 +41,77 @@ public static void beforeClass() throws Exception {
         StarRocksAssert starRocksAssert = new StarRocksAssert(connectContext);
         FeConstants.runningUnitTest = true;
         starRocksAssert.withTable("create table test(c0 INT, " +
-                "c1 struct<a array<struct<b int>>>," +
-                "c2 struct<a int,b double>) " +
+                "c1 struct<a int, b array<struct<a int, b int>>>," +
+                "c2 struct<a int,b double>," +
+                "c3 struct<a int, b int, c struct<a int, b int>>) " +
                 " duplicate key(c0) distributed by hash(c0) buckets 1 " +
                 "properties('replication_num'='1');");
         FeConstants.runningUnitTest = false;
     }
 
+    @Test
+    public void testSubfield() throws Exception {
+        String sql = "select c1.a from test";
+        String plan = getVerboseExplain(sql);
+        System.out.println(plan);
+    }
+
+    @Test
+    public void testSubfieldWindow() throws Exception {
+        String sql = "select c3.a, row_number() over (partition by c0 order by c2.a) from test";
+        String plan = getVerboseExplain(sql);
+        System.out.println(plan);
+    }
+
     @Test
     public void testSelectArrayStruct() throws Exception {
-        String sql = "select c1.a[10].b from test";
+        String sql = "select c1.b[10].a from test";
         String plan = getFragmentPlan(sql);
-        Assert.assertTrue(plan.contains("1:Project\n" +
-                "  |  <slot 4> : 2: c1.a[10].b"));
+        //        Assert.assertTrue(plan.contains("1:Project\n" +
```

Comment:
give right arrert

Reply:
@Smith-Cruise agree, better not to delete old case, instead change to rigth value.